packages/terra-date-picker/src/react-datepicker/index.jsx

```diff
@@ -0,0 +1,560 @@
+import Calendar from './calendar'
+import React from 'react'
+import PropTypes from 'prop-types'
+import PopperComponent, { popperPlacementPositions } from './popper_component'
+import classnames from 'classnames'
+import {
+  newDate,
+  now,
+  cloneDate,
+  isMoment,
+  isDate,
+  isBefore,
+  isAfter,
+  setTime,
+  getSecond,
+  getMinute,
+  getHour,
+  getMonth,
+  addDays,
+  addMonths,
+  addWeeks,
+  addYears,
+  subtractDays,
+  subtractMonths,
+  subtractWeeks,
+  subtractYears,
+  isSameDay,
+  isDayDisabled,
+  isDayInRange,
+  getEffectiveMinDate,
+  getEffectiveMaxDate,
+  parseDate,
+  safeDateFormat,
+  getHightLightDaysMap
+} from './date_utils'
+import onClickOutside from 'react-onclickoutside'
+
+const outsideClickIgnoreClass = 'react-datepicker-ignore-onclickoutside'
+const WrappedCalendar = onClickOutside(Calendar)
+
+/**
+ * General datepicker component.
+ */
+
+export default class DatePicker extends React.Component {
+  static propTypes = {
+    adjustDateOnChange: PropTypes.bool,
+    allowSameDay: PropTypes.bool,
+    autoComplete: PropTypes.string,
+    autoFocus: PropTypes.bool,
+    calendarClassName: PropTypes.string,
+    children: PropTypes.node,
+    className: PropTypes.string,
+    customInput: PropTypes.element,
+    customInputRef: PropTypes.string,
+    dateFormat: PropTypes.oneOfType([ // eslint-disable-line react/no-unused-prop-types
+      PropTypes.string,
+      PropTypes.array
+    ]),
+    dateFormatCalendar: PropTypes.string,
+    dayClassName: PropTypes.func,
+    disabled: PropTypes.bool,
+    disabledKeyboardNavigation: PropTypes.bool,
+    dropdownMode: PropTypes.oneOf(['scroll', 'select']).isRequired,
+    endDate: PropTypes.object,
+    excludeDates: PropTypes.array,
+    filterDate: PropTypes.func,
+    fixedHeight: PropTypes.bool,
+    formatWeekNumber: PropTypes.func,
+    highlightDates: PropTypes.array,
+    id: PropTypes.string,
+    includeDates: PropTypes.array,
+    inline: PropTypes.bool,
+    isClearable: PropTypes.bool,
+    locale: PropTypes.string,
+    maxDate: PropTypes.object,
+    minDate: PropTypes.object,
+    monthsShown: PropTypes.number,
+    name: PropTypes.string,
+    onBlur: PropTypes.func,
+    onChange: PropTypes.func.isRequired,
+    onSelect: PropTypes.func,
+    onWeekSelect: PropTypes.func,
+    onClickOutside: PropTypes.func,
+    onChangeRaw: PropTypes.func,
+    onFocus: PropTypes.func,
+    onKeyDown: PropTypes.func,
+    onMonthChange: PropTypes.func,
+    openToDate: PropTypes.object,
+    peekNextMonth: PropTypes.bool,
+    placeholderText: PropTypes.string,
+    popperContainer: PropTypes.func,
+    popperClassName: PropTypes.string, // <PopperComponent/> props
+    popperModifiers: PropTypes.object, // <PopperComponent/> props
+    popperPlacement: PropTypes.oneOf(popperPlacementPositions), // <PopperComponent/> props
```

Comment:
Which version of react-datepicker was this forked from? I was expecting there to be the `preventOpenOnFocus` prop as in [v1.8.0](https://github.com/Hacker0x01/react-datepicker/blob/v1.8.0/src/index.jsx#L127) that we could use in the future.

Reply:
Should be forked from v1.8.0, if it is not based on the v1.8.0 tag we should update before this is merged.

Reply:
You can disregard my comment of using 1.8.0 as the basis for the fork. Talked with @benbcai  and he mentioned he had explored updating terra-date-picker to use react-datepicker@1.8.0 and the UI crashed when interacting with the datepicker when using that version. He is going to post some more details. 

Based on this, I'd recommend we fork the [current version of react-datepicker that we're using today](https://github.com/cerner/terra-framework/blob/master/packages/terra-date-picker/package.json#L32). From talking with @JakeLaCombe, it sounds like that is the version used in this PR as the basis for the fork. Once this is merged, we can start doing work to cherry pick additional code changes from react-datepicker that we want to incorporate.

Reply:
I forked it based off of our current version in the repo of 0.64.0.

Reply:
If we were to fork V1.8.0, we would need to cherry pick the following PR:
- PR [#1548](https://github.com/Hacker0x01/react-datepicker/pull/1548) to fix issue  [#1547](https://github.com/Hacker0x01/react-datepicker/issues/1547). This was fixed in V2.1.0.

If we were to fork V0.64.0, we would need to cherry pick the following PRs:
- PR [#1181](https://github.com/Hacker0x01/react-datepicker/pull/1181) which introduces the `preventOpenOnFocus` prop that we would like to use.
- PR [#1374](https://github.com/Hacker0x01/react-datepicker/pull/1374) to remove unsafe react lifecycles.

V1.8.0 is obviously more updated than V0.64.0. But since we don't know what other broken changes (other than [#1547](https://github.com/Hacker0x01/react-datepicker/issues/1547))  we would run into if we were to use 1.8.0, I'm fine forking V0.64.0 and then subsequently pull in these additional pull requests that we need.